kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/OperationContext.java

```diff
@@ -39,7 +39,8 @@
   protected boolean cascading;
   protected boolean reloadingFromServer;
 
-  protected long gracePeriodSeconds;
+  // Default to k8s 30s value: https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods
+  protected long gracePeriodSeconds = 30L;
```

Comment:
I run into this issue today when I found out that when deleting pods the termination grace period in my pods is always overwritten to `0` and the pods are basically force deleted. 

However, I'm wondering if this is the right fix. It still overwrites whatever the user configured in his pod which is IMHO wrong. I think the default should be -1 which will mean that the gracePeriod is not set in the request and the default from the resource will be used by Kubernetes => that will be either 30 seconds if the user didn't specified anything or whatever the user specified in his pods.

Reply:
haven't had time to test the `grace period` to `-1`, but, if it works, I see your concern and I agree on the approach.

Reply:
this PR got already merged, do you mind to follow up?

Reply:
I opened #1833 to change the default grace period value.